client/src/connection/rest/index.ts

```diff
@@ -54,10 +55,33 @@ async function setup() {
 
   if (config.serverId) {
     const server1 = new ComputeServer(config.serverId);
+
+    //if we have a session already, get it
```

Comment:
It looks like only do if serverId exists. What about no serverId? I think it also makes sense to reconnect, right?

Reply:
I had not tested this feature in a Viya environment, so I limited the reconnection to the serverId case for now. I planned on revisiting the Viya case after discussion on redesign of the Session interface.